src/Behat/Gherkin/Parser.php

```diff
@@ -262,14 +264,12 @@ protected function parseFeature()
                 ));
             }
 
-            if (!$node instanceof ScenarioNode) {
```

Comment:
This condition and the other cases below were flagged as impossible by PHPStan because:
1. there is already an earlier branch unconditionally handling such instance
2. `$node` is never overwritten